enclave-runtime/src/tls_ra/tls_ra_client.rs

```diff
@@ -239,9 +248,45 @@ pub unsafe extern "C" fn request_state_provisioning(
 		return e.into()
 	};
 
+	if let Err(e) = touch_shard(shard) {
+		error!("touch shard error: {:?}", e);
+		return sgx_status_t::SGX_ERROR_UNEXPECTED
+	}
 	sgx_status_t::SGX_SUCCESS
 }
 
+fn touch_shard(shard: ShardIdentifier) -> EnclaveResult<()> {
+	// send confirmation about provisioning to chain to signal that we're ready to serve the shard (this will not yield an event because secondary validateers are ignored.)
+	// fixme: it would be more elegant to have a separate dispatchable for this like `touch_shard` so we don't need to abuse this call
+	// https://github.com/integritee-network/pallets/issues/232
```

Comment:
true, we could let the secondary call the primary rpc to tell "I'm here" after they registered their enclave
but then: they already requested provisioning, so the primary could know them already

I agree we should question the approach, but I think this will be reworked when introducing ShardConfig with permissioned or limited set of validateers anyway, so I'd like to not polish this now